This file is part of MXE.
See index.html for further information.

Avoid DllMain symbol conflict when linking statically.

diff -uNr glib-2.28.8-orig/gio/giomodule.c glib/gio/giomodule.c
--- glib-2.28.8-orig/gio/giomodule.c	2011-01-05 22:44:06.000000000 +0100
+++ glib/gio/giomodule.c	2011-11-23 21:04:43.867004895 +0100
@@ -489,10 +489,8 @@
 
 static HMODULE gio_dll = NULL;
 
-#ifdef DLL_EXPORT
-
 BOOL WINAPI
-DllMain (HINSTANCE hinstDLL,
+gio_DllMain (HINSTANCE hinstDLL,
 	 DWORD     fdwReason,
 	 LPVOID    lpvReserved)
 {
@@ -502,8 +500,6 @@
   return TRUE;
 }
 
-#endif
-
 #undef GIO_MODULE_DIR
 
 /* GIO_MODULE_DIR is used only in code called just once,
diff --git a/glib/glib-init.c b/glib/glib-init.c
index f4edd5c..02bf678 100644
--- a/glib/glib-init.c
+++ b/glib/glib-init.c
@@ -226,9 +226,9 @@ glib_init (void)
 HMODULE glib_dll;
 
 BOOL WINAPI
-DllMain (HINSTANCE hinstDLL,
-         DWORD     fdwReason,
-         LPVOID    lpvReserved)
+glib_DllMain (HINSTANCE hinstDLL,
+              DWORD     fdwReason,
+              LPVOID    lpvReserved)
 {
   switch (fdwReason)
     {
